data_iterator.go

```diff
@@ -236,7 +100,7 @@ func (d *DataIterator) Run() {
 				}
 
 				logger.Debug("table iteration completed")
-				d.CurrentState.MarkTableAsCompleted(table.String())
+				d.StateTracker.MarkTableAsCompleted(table.String())
```

Comment:
I love how all call sites to the critical StateTracker methods make sense now. I think we did good 🙂